.github/workflows/pr-arm.yml

```diff
@@ -0,0 +1,102 @@
+name: "PR Tests (aarch64)"
+permissions: read-all
+
+# Trigger for PR and merge to develop branch
+on:
+  push:
+    branches: develop
+  pull_request:
+  workflow_dispatch:
+
+env:
+  CTEST_OUTPUT_ON_FAILURE: 1
+  LAPACK_VERSION: 3.12.0
+  ARMPL_VERSION: 25.04.1
+  DPCPP_VERSION: nightly-2025-04-01
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-24.04-arm
+    # One runner for each domain
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - config: oneMath BLAS
+          domain: blas
+          build_options: -DREF_BLAS_ROOT=${PWD}/lapack/install
+        - config: oneMath LAPACK
+          domain: lapack
+          build_options: -DREF_LAPACK_ROOT=${PWD}/lapack/install
+        - config: oneMath RNG
+          domain: rng
+          test_options: -E 'Device|DEVICE'
+    name: unit tests ${{ matrix.config }} CPU
+    steps:
+    - uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+    - name: Check if the changes affect this domain
+      id: domain_check
+      uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+      with:
+        script: |
+         const domainCheck = require('.github/scripts/domain-check.js')
+         return domainCheck({github, context, domain: "${{ matrix.domain }}"})
+    - name: Restore netlib from cache
+      id: cache-lapack
+      uses: actions/cache@d4323d4df104b026a6aa633fdb11d772146be0bf # v4.2.2
+      with:
+        path: lapack/install
+        key: lapack-${{ env.LAPACK_VERSION }}-arm
```

Comment:
I am not sure if this key will work. Could you please also check if the cache for compiler is working?